### ISSUE TITLE
Support Java 11

### ISSUE THREAD

**Issue by Quaiks** [jenkinsci/templating-engine-plugin#28]
Hello!!

I am trying to run the plugin in a Java 11 Jenkins instance and I am getting the following error

```java
java.lang.ClassNotFoundException: @63a800a7)
 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
 	at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:677)
 	at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:787)
 	at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:775)
 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
 	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:47)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
 	at org.boozallen.plugins.jte.utils.TemplateScriptEngine.parseClass(TemplateScriptEngine.groovy:76)
 	at org.boozallen.plugins.jte.utils.TemplateScriptEngine$parseClass.call(Unknown Source)
 	at org.boozallen.plugins.jte.binding.injectors.LibraryLoader.getPrimitiveClass(LibraryLoader.groovy:95)
 	at org.boozallen.plugins.jte.binding.injectors.LibraryLoader$getPrimitiveClass.call(Unknown Source)
 	at org.boozallen.plugins.jte.binding.TemplateBinding.setVariable(TemplateBinding.groovy:43)
 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
 	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133)
 	at org.boozallen.plugins.jte.TemplateEntryPointVariable.getValue(TemplateEntryPointVariable.groovy:65)
 	at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:113)
 	at jdk.internal.reflect.GeneratedMethodAccessor1079.invoke(Unknown Source)
 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
 	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:42)
 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
 	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:160)
 	at org.kohsuke.groovy.sandbox.GroovyInterceptor.onMethodCall(GroovyInterceptor.java:23)
 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:157)
 	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onMethodCall(SandboxInterceptor.java:142)
 	at org.kohsuke.groovy.sandbox.impl.Checker$1.call(Checker.java:158)
 	at org.kohsuke.groovy.sandbox.impl.Checker.checkedCall(Checker.java:162)
 	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.methodCall(SandboxInvoker.java:17)
 	at WorkflowScript.run(WorkflowScript:1)
 	at ___cps.transform___(Native Method)
 	at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:84)
 	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:113)
 	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixName(FunctionCallBlock.java:78)
 	at jdk.internal.reflect.GeneratedMethodAccessor761.invoke(Unknown Source)
 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
 	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
 	at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
 	at com.cloudbees.groovy.cps.Next.step(Next.java:83)
 	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
 	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
 	at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:129)
 	at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:268)
 	at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
 	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
 	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:51)
 	at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:186)
 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:370)
 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$200(CpsThreadGroup.java:93)
 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:282)
 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:270)
 	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:66)
 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
 	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
 	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 	at java.base/java.lang.Thread.run(Thread.java:834)
```

I think the problem is related with Java 11.

Thanks in advance!

**Comment by steven-terrana** [comment]
we'll do some testing, see what we can find, and get back to you! 

Our testing is primarily done against the community jenkins image jenkins/jenkins:lts

we'll have to expand our test cases :) 

thank you!

**Comment by steven-terrana** [comment]
hey @Quaiks - we've identified and resolved this issue in #31. 

will release 1.4.1 with the fix shortly.

**Comment by Quaiks** [comment]
Nice!!
